cognite/neat/rules/models/_rules/_validators/_scenario.py

```diff
@@ -0,0 +1,33 @@
+from copy import deepcopy
+from typing import cast
+
+from cognite.neat.rules._analysis import DataModelingScenario, InformationArchitectRulesAnalysis
+from cognite.neat.rules.issues import IssueList, handle_issues
+from cognite.neat.rules.models._rules import InformationRules
+from cognite.neat.rules.models._rules.base import SchemaCompleteness
+
+from ._base import BaseRulesValidator
+
+
+class DataModelingScenarioValidator(BaseRulesValidator):
+    def validate(self) -> IssueList:
+        if isinstance(self.rules, InformationRules):
+            return self._validate_information_rules()
+        else:
+            return self.issue_list
+
+    def _validate_information_rules(self) -> IssueList:
+        analysis = InformationArchitectRulesAnalysis(cast(InformationRules, self.rules))
+        if analysis.data_modeling_scenario == DataModelingScenario.build_solution:
+            print("Here")
+            cp_rules = deepcopy(cast(InformationRules, self.rules))
```

Comment:
pydantic has a built in deep copy 
```suggestion
            cp_rules = cast(InformationRules, self.rules)).copy(deep=True)
```

And using generics in the base class, saves you from having to do the cast.